File: scripts/python/file_operations.py
#!/usr/bin/env python3
"""
File operations example script.
Demonstrates reading, writing, and processing files.
"""
import os
import json


def read_file(filename):
    """Read and return contents of a file."""
    with open(filename, 'r') as f:
        return f.read()


def write_file(filename, content):
    """Write content to a file."""
    with open(filename, 'w') as f:
        f.write(content)


def read_json(filename):
    """Read and parse a JSON file."""
    with open(filename, 'r') as f:
        return json.load(f)


def write_json(filename, data):
    """Write data to a JSON file."""
    with open(filename, 'w') as f:
        json.dump(data, f, indent=2)


def main():
    """Demonstrate file operations."""
    # Example data
    data = {
        "name": "Test Data",
        "values": [1, 2, 3, 4, 5],
        "active": True
    }
    
    print("File Operations Example")
    print(f"Sample data: {data}")


if __name__ == "__main__":
    main()

File: scripts/javascript/file_operations.js
#!/usr/bin/env node
/**
 * File operations example script.
 * Demonstrates reading, writing, and processing files.
 */
const fs = require('fs').promises;
const path = require('path');

async function readFile(filename) {
  return await fs.readFile(filename, 'utf-8');
}

async function writeFile(filename, content) {
  await fs.writeFile(filename, content, 'utf-8');
}

async function readJSON(filename) {
  const content = await readFile(filename);
  return JSON.parse(content);
}

async function writeJSON(filename, data) {
  const content = JSON.stringify(data, null, 2);
  await writeFile(filename, content);
}

async function main() {
  console.log('File Operations Example');
  
  const data = {
    name: 'Test Data',
    values: [1, 2, 3, 4, 5],
    active: true
  };
  
  console.log('Sample data:', data);
}

if (require.main === module) {
  main().catch(console.error);
}

module.exports = { readFile, writeFile, readJSON, writeJSON };
